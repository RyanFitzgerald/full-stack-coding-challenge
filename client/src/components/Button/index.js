import styled from 'styled-components';

const Button = styled.button`
  background: #3498db;
  border: 0;
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  display: block;
  margin: 15px auto 0;
  padding: 15px 50px;
  transition: 0.25s ease all;

  &:hover {
    background: #236c9c;
  }
`;

export default Button;