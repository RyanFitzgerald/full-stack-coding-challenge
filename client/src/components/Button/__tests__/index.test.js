import React from 'react';

import Button from '../index';

describe('<Button />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <Button>Test</Button>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct element', () => {
    const renderedComponent = shallow(
      <Button/>
    );
    
    expect(renderedComponent.type()).toEqual('button');
  });

  it('should render its content', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <Button>{children}</Button>
    );
    
    expect(renderedComponent.contains(children)).toBe(true);
  });
});