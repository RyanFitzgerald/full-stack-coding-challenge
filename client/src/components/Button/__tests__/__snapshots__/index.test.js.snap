// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
.c0 {
  background: #3498db;
  border: 0;
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  display: block;
  margin: 15px auto 0;
  padding: 15px 50px;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
}

.c0:hover {
  background: #236c9c;
}

<button
  className="c0"
>
  Test
</button>
`;
